src/main/java/com/aws/greengrass/clientdevices/auth/configuration/CAConfiguration.java

```diff
@@ -114,6 +114,16 @@ private static CertificateStore.CAType getCaTypeFromConfiguration(Topics configu
         return CertificateStore.CAType.valueOf(caType);
     }
 
+    private static URI getUriOrThrow(String maybeUri) throws URISyntaxException {
```

Comment:
Calling `new URI` with a URI string without scheme is valid but would explode down the line. The uri we ask for must have a scheme

Reply:
I like it. It's easy to relax restrictions but hard to add them